### PR TITLE
Anasmunir92/dev 502 checkout steps stripe checkout

### PR DESCRIPTION
**Pull request by AnasMunir**
A collection of task for the entire checkout flow

### CODE REVIEW

File: src/components/Checkout/MemberFrequencyPlan.tsx [hunk 1]
```diff
@@ -9,35 +9,64 @@ import { splitList } from '@/helpers/misc';
...
+		params.set('membership', offerings[activeTabIdx].id);
+		if (window) {
+			window.history.pushState(null, '', `?${params.toString()}`);
+			window.scrollTo({ top: 0 });
+		}
 		// TODO: remove console and go to payment link
-		console.log(selectedPayment, emailUser, product); // eslint-disable-line no-console
+		if (setStep) {
+			setStep(CheckoutStep.STRIPE_PAYMENT);
+			router.push(`/onboarding/payment?product=${product}&&membership=${offerings[activeTabIdx].id}`);
```
Comment (RidhoIdris) [inline src/components/Checkout/MemberFrequencyPlan.tsx:68]: double &&

File: src/components/Checkout/Form.tsx [hunk 2]
```diff
@@ -249,13 +288,15 @@ const Form: React.FC<{
...
+									className='max-sm:w-full bg-blue-primary text-primary py-[17px] px-[42px] rounded-full hover:brightness-105 transform transition duration-200 focus:ring-0 focus:outline-none'
+								>
 									{ formSectionData.submitLabel }
 								</button>
 								<button
 									type='button'
-									onClick={ () => router.back() }
-									className='max-sm:w-full text-grey-primary border border-grey-primary transition duration-200 hover:brightness-105 py-[17px] px-[42px] rounded-full focus:ring-0 focus:outline-none'>
+									onClick={ () => router.replace('/') }
+									className='max-sm:w-full text-grey-primary border border-grey-primary transition duration-200 hover:brightness-105 py-[17px] px-[42px] rounded-full focus:ring-0 focus:outline-none'
```
Comment (RidhoIdris) [inline src/components/Checkout/Form.tsx:298]: i think we already have a button component

File: src/components/Checkout/StripeCheckout/DiscountForm.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+				disabled={ loading }
+			/>
+			<div className='absolute w-4 h-4 lg:w-6 lg:h-6 right-[22px] bottom-[22px] lg:bottom-[18px]'>
+				{ discountApplied && <GreenCircleTick /> }
+			</div>
+		</form>
+	);
+};
+
+function GreenCircleTick() {
```
Comment (RidhoIdris) [inline src/components/Checkout/StripeCheckout/DiscountForm.tsx:52]: move to icon component

File: src/components/Checkout/StripeCheckout/StripeCheckout.tsx [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+					loading={ loading }
+					totalPrice={ totalPrice }
+					handleCheckout={ handleCheckout }
+				/>
+			</div>
+		</div>
+	);
+};
+
+interface ITotalCalc {
```
Comment (RidhoIdris) [inline src/components/Checkout/StripeCheckout/StripeCheckout.tsx:193]: move to the interface file

File: src/components/Checkout/StripeCheckout/StripeForm.tsx [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+	CardExpiryElement,
+	CardNumberElement,
+} from '@stripe/react-stripe-js';
+import { Stripe, StripeElements, StripeError } from '@stripe/stripe-js';
+import { toast } from 'sonner';
+
+import Checkbox from '@/components/Onboarding/Checkbox';
+import clsxm from '@/helpers/clsxm';
+
+type StripeFormProps = {
```
Comment (RidhoIdris) [inline src/components/Checkout/StripeCheckout/StripeForm.tsx:13]: move to the interface file

File: src/components/Checkout/StripeCheckout/StripeForm.tsx [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+					</button>
+				</div>
+			</div>
+		</form>
+	);
+};
+
+export default StripeForm;
+
+function Spinner() {
```
Comment (RidhoIdris) [inline src/components/Checkout/StripeCheckout/StripeForm.tsx:145]: move to the icons component

File: src/components/Checkout/StripeCheckout/StripeForm.tsx (hunk 5 above)
Comment (AnasMunir) [reply src/components/Checkout/StripeCheckout/StripeForm.tsx:13]: These are component props, they can stay here

File: src/components/Checkout/Form.tsx (hunk 2 above)
Comment (AnasMunir) [reply src/components/Checkout/Form.tsx:298]: Code updated

File: src/components/Checkout/StripeCheckout/StripeCheckout.tsx (hunk 4 above)
Comment (AnasMunir) [reply src/components/Checkout/StripeCheckout/StripeCheckout.tsx]: Separated into a component

**Pull request merged**